persist/fs/read.go

```diff
@@ -63,43 +67,164 @@ type reader struct {
 	blockSize      time.Duration
 
 	infoFdWithDigest           digest.FdWithDigestReader
-	indexFdWithDigest          digest.FdWithDigestReader
-	dataFdWithDigest           digest.FdWithDigestReader
 	digestFdWithDigestContents digest.FdWithDigestContentsReader
-	expectedInfoDigest         uint32
-	expectedIndexDigest        uint32
-	expectedDataDigest         uint32
-	expectedDigestOfDigest     uint32
-
-	unreadBuf   []byte
-	entries     int
-	entriesRead int
-	prologue    []byte
-	decoder     encoding.Decoder
-	digestBuf   digest.Buffer
-	bytesPool   pool.CheckedBytesPool
+
+	indexFd            *os.File
+	indexMmap          []byte
+	indexDecoderStream *decoderStream
+
+	dataFd     *os.File
+	dataMmap   []byte
+	dataReader digest.ReaderWithDigest
+
+	expectedInfoDigest     uint32
+	expectedIndexDigest    uint32
+	expectedDataDigest     uint32
+	expectedDigestOfDigest uint32
+	entries                int
+	entriesRead            int
+	metadataRead           int
+	prologue               []byte
+	decoder                encoding.Decoder
+	digestBuf              digest.Buffer
+	bytesPool              pool.CheckedBytesPool
+
+	indexEntriesByOffsetAsc []schema.IndexEntry
+}
+
+type indexEntriesByOffsetAsc []schema.IndexEntry
```

Comment:
nit: move `indexEntriesByOffsetAsc` and `decoderStream` to the bottom of the file

Reply:
Sure thing.